add a helper to match a rcf822 datetime string

cabm opened the pull request.

bethesque commented:
Hey @cabm, can you tell me more about this please. Firstly, we need to ensure that it's actually a date format used by many of the pact users. Secondly, we need a method name that actually identifies what the date format is to everyone using it - the term "non iso861" is very broad. Thirdly, we'd need more test cases! Keep in mind that pact is used across many technology stacks, so when something goes into the library, we need to consider a lot of different users and languages.

cabm commented:
Thanks @bethesque for the feedback. I changed the name and added more tests.  If there were other test scenarios you were looking for let me know. I believe this format is what Rails defaults to when generating a DateTime object.  Let me know if you have any other questions/concerns. Cheers, Courtney

bethesque commented:
Thanks @cabm.
